src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -19,38 +19,50 @@
 from pants.backend.python.targets.python_tests import PythonTests
 from pants.base.build_environment import get_buildroot
 from pants.base.exceptions import TaskError
-from pants.build_graph.resources import Resources
+from pants.build_graph.files import Files
 from pants.python.python_repos import PythonRepos
 
 
 def has_python_sources(tgt):
   # We'd like to take all PythonTarget subclasses, but currently PythonThriftLibrary and
   # PythonAntlrLibrary extend PythonTarget, and until we fix that (which we can't do until
   # we remove the old python pipeline entirely) we want to ignore those target types here.
-  return isinstance(tgt, (PythonLibrary, PythonTests, PythonBinary, Resources))
+  return isinstance(tgt, (PythonLibrary, PythonTests, PythonBinary)) and tgt.has_sources()
 
 
 def has_resources(tgt):
-  return isinstance(tgt, Resources)
+  return isinstance(tgt, Files) and tgt.has_sources()
 
 
 def has_python_requirements(tgt):
   return isinstance(tgt, PythonRequirementLibrary)
 
 
-def dump_sources(builder, tgt, log):
+def _create_source_dumper(builder, tgt):
+  if type(tgt) == Files:
+    # Loose `Files` as opposed to `Resources` or `PythonTarget`s have no (implied) package structure
+    # and so we chroot them relative to the build root so that they can be accessed via the normal
+    # python filesystem APIs just as they would be accessed outside the chrooted environment.
+    # NB: This requires we mark the pex as not zip safe so these `Files` can still be accessed in
+    # the context of a built pex distribution.
+    chroot_path = lambda relpath: relpath
```

Comment:
Did you consider changing `Files` to have its own `sources_relative_to_source_root` that just uses the buildroot? I think that would simplify this somewhat and allow other tasks to refer to Files in a consistent way. It could be that I'm missing something though.

Reply:
I didn't. Thinking about that idea now though, I'm not sure I like it. Code would simplify, but it would read misleadingly since a source root just doesn't make sense for loose file system access. Ideally `Files.sources_relative_to_source_root` would raise, more ideally `Target` wouldn't house that method since it doesn't make sense for all `Target` subclasses. The misleading bit aside, you're of course right that handling `Files` sources differently from `Resources` sources is up to each task all over again which may not be ideal. Even though I want to say the concept of `Resources` may be language specific, I can't think of an example language that wouldn't go through the same hoops here.